Lib/threading.py

```diff
@@ -1376,6 +1397,12 @@ def _shutdown():
     tlock.release()
     _main_thread._stop()
 
+    # Call registered threading atexit functions before threads are joined
+    if _threading_atexits is not None:
+        # Order is reversed, similar to atexit.
+        for atexit_call in reversed(_threading_atexits):
```

Comment:
I wonder if it's worth checking if one of the atexit callbacks didn't call `_register_atexit`?

Reply:
That's a good point that I hadn't considered. I personally don't see a strong realistic use case for having to register another atexit callback with an atexit callback, so I think that can be "unsupported" for now (until someone comes up with one).

I'm not certain what the behavior should be though. Would it simply be ignored, raise an exception, or something else? I'm leaning towards an exception (to be explicit), but I could see that causing issues with the shutdown process. 

Perhaps there could be a global flag that gets set to true before calling the atexits in `_shutdown()` such as "`_SHUTTING_DOWN`, which would be checked in `_register_atexit()`, prior to registering the callback.

Reply:
Yes, you could e.g. raise an exception there.

Reply:
Done.